plugins/header_rewrite/value.cc

```diff
@@ -63,13 +63,20 @@ Value::set_value(const std::string &val)
           }
         }
       } else {
+        if (token.find("$") != std::string::npos && p.get_regex_cond().length() > 0) {
+          _regex_cond = condition_factory(p.get_regex_cond());
+          _regex_pat  = p.get_regex_pat();
+        }
         tcond_val = new ConditionStringLiteral(token);
       }
 
       if (tcond_val) {
         _cond_vals.push_back(tcond_val);
       }
     }
+  } else if (_value.find("$") != std::string::npos && p.get_regex_cond().length() > 0) {
```

Comment:
Same here, check if the regex condition exists first, before doing the string search ?